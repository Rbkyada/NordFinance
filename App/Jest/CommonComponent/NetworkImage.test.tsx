import renderer from 'react-test-renderer';
import { NetworkImage } from '@CommonComponent/NetworkImage';

describe('NetworkImage component', () => {
  it('renders correctly with valid source', () => {
    const source = 'https://example.com/image.jpg';
    const tree = renderer.create(<NetworkImage source={source} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders placeholder when isLoading is true', () => {
    const source = 'https://example.com/image.jpg';
    const placeholder = 'https://example.com/placeholder.jpg';
    const tree = renderer
      .create(<NetworkImage source={source} placeholder={placeholder} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders error image when isError is true', () => {
    const source = 'https://example.com/non-existing-image.jpg';
    const errorImage = 'https://example.com/error-image.jpg';
    const tree = renderer
      .create(<NetworkImage source={source} errorImage={errorImage} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders loading container while loading', () => {
    const source = 'https://example.com/image.jpg';
    const tree = renderer.create(<NetworkImage source={source} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom styles', () => {
    const source = 'https://example.com/image.jpg';
    const containerStyle = { borderRadius: 10 };
    const imageStyle = { width: 200, height: 200 };
    const tree = renderer
      .create(
        <NetworkImage
          source={source}
          containerStyle={containerStyle}
          imageStyle={imageStyle}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
