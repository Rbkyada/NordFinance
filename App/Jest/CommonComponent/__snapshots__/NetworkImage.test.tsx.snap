// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkImage component renders correctly with custom styles 1`] = `
<View
  style={
    [
      {
        "height": 100,
        "width": 100,
      },
      {
        "height": 200,
        "width": 200,
      },
      {
        "borderRadius": 10,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "height": 100,
            "width": 100,
          },
          {
            "height": 200,
            "width": 200,
          },
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 100,
          "justifyContent": "center",
          "position": "absolute",
          "width": 100,
        },
        {
          "height": 200,
          "width": 200,
        },
        {
          "borderRadius": 10,
        },
      ]
    }
  >
    <ActivityIndicator />
  </View>
</View>
`;

exports[`NetworkImage component renders correctly with valid source 1`] = `
<View
  style={
    [
      {
        "height": 100,
        "width": 100,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "height": 100,
            "width": 100,
          },
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 100,
          "justifyContent": "center",
          "position": "absolute",
          "width": 100,
        },
        undefined,
        undefined,
      ]
    }
  >
    <ActivityIndicator />
  </View>
</View>
`;

exports[`NetworkImage component renders error image when isError is true 1`] = `
<View
  style={
    [
      {
        "height": 100,
        "width": 100,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "height": 100,
            "width": 100,
          },
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/non-existing-image.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 100,
          "justifyContent": "center",
          "position": "absolute",
          "width": 100,
        },
        undefined,
        undefined,
      ]
    }
  >
    <ActivityIndicator />
  </View>
</View>
`;

exports[`NetworkImage component renders loading container while loading 1`] = `
<View
  style={
    [
      {
        "height": 100,
        "width": 100,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "height": 100,
            "width": 100,
          },
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 100,
          "justifyContent": "center",
          "position": "absolute",
          "width": 100,
        },
        undefined,
        undefined,
      ]
    }
  >
    <ActivityIndicator />
  </View>
</View>
`;

exports[`NetworkImage component renders placeholder when isLoading is true 1`] = `
<View
  style={
    [
      {
        "height": 100,
        "width": 100,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "height": 100,
            "width": 100,
          },
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "height": 100,
          "justifyContent": "center",
          "position": "absolute",
          "width": 100,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Image
      source={
        {
          "uri": "https://example.com/placeholder.jpg",
        }
      }
      style={
        [
          {
            "height": 100,
            "width": 100,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
